packages/format-library/src/link/index.js

```diff
@@ -56,10 +74,11 @@ function Edit( {
 			// to be rendered in "creating" mode. We need to check isActive to see if
 			// we have an active link format.
 			if ( event.target.tagName !== 'A' || ! isActive ) {
+				setIsEditingExistingLink( false );
 				return;
 			}
 
-			setAddingLink( true );
+			setIsEditingExistingLink( true, { shouldAutoFocus: false } );
```

Comment:
```suggestion
			setIsEditingExistingLink( ( curVal ) => ! curVal, {
				shouldAutoFocus: false,
			} );
```

If we want to make this PR do the "click to toggle" behaviour we can commit this change.